mozetl/search/dashboard.py

```diff
@@ -0,0 +1,113 @@
+import click
+import logging
+from pyspark.sql.functions import explode, col, when
+from pyspark.sql import SparkSession
+
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+def search_dashboard_etl(main_summary):
+    # todo: this function should consume already exploded and augmented data
+    exploded = explode_search_counts(main_summary)
+    augmented = add_derived_columns(exploded)
+    group_cols = filter(lambda x: x != 'count', augmented.columns)
+
+    return (
+        augmented
+        .groupBy(group_cols)
+        .sum('count')
+        .withColumnRenamed('sum(count)', 'search_count')
+    )
+
+
+def explode_search_counts(main_summary):
+    input_columns = [
+        'search_counts',
+        'submission_date',
+        'country',
+        'app_version',
+        'distribution_id',
+    ]
+    exploded_col_name = 'single_search_count'
+
+    def define_sc_column(field):
+        return field, col(exploded_col_name + '.' + field)
+
+    return (
+        main_summary
+        .select(input_columns)
+        .withColumn(exploded_col_name, explode(col('search_counts')))
+        .withColumn(*define_sc_column('engine'))
+        .withColumn(*define_sc_column('source'))
+        .withColumn(*define_sc_column('count'))
+        .drop(exploded_col_name)
+        .drop('search_counts')
+    )
+
+
+def add_derived_columns(exploded_search_counts):
```

Comment:
Add a more descriptive function name like `with_derived_search_type` or a comment related to the type of derived column that it adds to the dataframe.